src/VirtualClient/VirtualClient.Actions/Memtier/MemtierBenchmarkClientExecutor.cs

```diff
@@ -346,6 +368,85 @@ private void CaptureMetrics(string results, string commandArguments, DateTime st
             }
         }
 
+        private void CaptureAggregateMetrics(string commandArguments, DateTime startTime, DateTime endTime, EventContext telemetryContext, CancellationToken cancellationToken)
```

Comment:
Can you do the aggregation in the parser class? This allows us to properly unit test it

Reply:
Here we aren't using results/metrics from single object of Parser and aggregating the metrics. We are aggregating the metrics across results of different parser objects.

Reply:
You can still do it with Parser, the constructor is free, you can change that to params[] string or oput the Aggregate method on Parser, or something else